src/components/table/helpers/mixin-thead.js

```diff
@@ -100,23 +100,8 @@ export default {
           on: handlers
         }
         const fieldScope = { label: field.label, column: field.key, field, isFoot }
-        let slot
-        if (
-          isFoot &&
-          this.hasNormalizedSlot([`FOOT[${field.key}]`, 'FOOT[]', `FOOT_${field.key}`])
-        ) {
-          // TODO: `FOOT_${field.key}` is deprecated, to be removed in future release
-          slot = this.normalizeSlot(
-            [`FOOT[${field.key}]`, 'FOOT[]', `FOOT_${field.key}`],
-            fieldScope
-          )
-        } else {
-          // TODO: `HEAD_${field.key}` is deprecated, to be removed in future release
-          slot = this.normalizeSlot(
-            [`HEAD[${field.key}]`, 'HEAD[]', `HEAD_${field.key}`],
-            fieldScope
-          )
-        }
+        const slotKey = isFoot ? 'FOOT' : 'HEAD'
+        const slot = this.normalizeSlot([`${slotKey}[${field.key}]`, `${slotKey}[]`], fieldScope)
```

Comment:
This still needs to be able to fallback, for footer, to use the `HEAD` slots if a `FOOT` slot is not provided.

```js
        if (isFoot && this.hasNormalizedSlot([`FOOT[${field.key}]`, 'FOOT[]')) {
          slot = this.normalizeSlot([`FOOT[${field.key}]`, 'FOOT[]'],  fieldScope)
        } else {
          slot = this.normalizeSlot([`HEAD[${field.key}]`, 'HEAD[]'], fieldScope)
        }
```